pkg/kubeadm/master/manifests.go

```diff
@@ -0,0 +1,194 @@
+
+func WriteStaticPodManifests(params *kubeadmapi.BootstrapParams) error {
+	staticPodSpecs := map[string]api.Pod{
+		// TODO this needs a volume
+		"etcd": componentPod(api.Container{
+			Command: []string{
+				"/usr/local/bin/etcd",
+				"--listen-peer-urls=http://127.0.0.1:2380",
+				"--addr=127.0.0.1:2379",
+				"--bind-addr=127.0.0.1:2379",
+				"--data-dir=/var/etcd/data",
+			},
+			Image:         "gcr.io/google_containers/etcd:2.2.1", // TODO parametrise
+			LivenessProbe: componentProbe(2379, "/health"),
+			Name:          "etcd-server",
+			Resources:     componentResources("200m"),
+		}),
+		// TODO bind-mount certs in
+		"kube-apiserver": componentPod(api.Container{
+			Name:  "kube-apiserver",
+			Image: params.EnvParams["hyperkube_image"],
+			Command: []string{
+				"/hyperkube",
+				"apiserver",
+				"--address=127.0.0.1",
+				"--etcd-servers=http://127.0.0.1:2379",
+				"--cloud-provider=fake", // TODO parametrise
+				"--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,ResourceQuota",
+				SERVICE_CLUSTER_IP_RANGE,
+				"--service-account-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--client-ca-file=/etc/kubernetes/pki/ca.pem",
+				"--tls-cert-file=/etc/kubernetes/pki/apiserver.pem",
+				"--tls-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--secure-port=443",
+				"--allow-privileged",
+				COMPONENT_LOGLEVEL,
+				"--token-auth-file=/etc/kubernetes/pki/tokens.csv",
+			},
+			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
+			LivenessProbe: componentProbe(8080, "/healthz"),
+			Resources:     componentResources("250m"),
+		}, pkiVolume(params)),
+		"kube-controller-manager": componentPod(api.Container{
+			Name:  "kube-controller-manager",
+			Image: params.EnvParams["hyperkube_image"],
+			Command: []string{
+				"/hyperkube",
+				"controller-manager",
+				MASTER,
+				CLUSTER_NAME,
+				"--root-ca-file=/etc/kubernetes/pki/ca.pem",
+				"--service-account-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--cluster-signing-cert-file=/etc/kubernetes/pki/ca.pem",
+				"--cluster-signing-key-file=/etc/kubernetes/pki/ca-key.pem",
+				"--insecure-experimental-approve-all-kubelet-csrs-for-group=system:kubelet-bootstrap",
+				COMPONENT_LOGLEVEL,
```

Comment:
One thing docker does well is a bunch of heuristic-y things to find appropriate network ranges for stuff like this.  May be worth looking there for inspiration.

When we do "guess" stuff like this we should print it out as part of the output of `kubeadm init` along with a note that it can be changed with a command line flag.